### PR TITLE
Add Sapphire docker image

### PR DESCRIPTION
**Pull request by matevz**
Docker images running complete oasis stack including our ParaTimes on a Localnet are useful for our dApp developers both for e2e tests on their CI and for local development, where Testnet is not appropriate for whichever reason.

This PR:
- refactors `spinup-oasis-stack.sh` to support Sapphire
- adds `sapphire-dev` docker image to CI
- refactors Docker scripts to use common docker parts as possible (server.yml, oasis-deposit, start.sh)

Merge after emerald (https://github.com/oasisprotocol/emerald-paratime/pull/27) and sapphire (https://github.com/oasisprotocol/sapphire-paratime/pull/102) Localnet PRs are merged and releases deployed.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+## Usage
+
+Once the docker image is up and running, you can connect your hardhat,
+truffle or metamask the *localnet* running at `http://localhost:8545` and
+`ws://localhost:8546`.
+
+Localnet chain IDs:
+- Emerald: `0xa5i4` (`42260`)
```
Comment (gw0) [inline docker/README.md]: `0xa5i4` is not a valid hexadecimal value. Also, to not confuse developers use full network+paratime names when referring to non-Mainnet (e.g. Localnet Emerald).

**Review by nhynes — APPROVED** [review]

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+set up localnet for developing dApps running on Emerald or Sapphire:
+
+- oasis-node and oasis-net-runner
+- Emerald or Sapphire ParaTime .orc bundles
+- oasis-web3-gateway
+- oasis-deposit helper to fund local development accounts
+
+## Prebuilt images
+
+Oasis provides prebuilt `emerald-dev` and `sapphire-dev` docker images. `latest`
```
Comment (ptrus) [inline docker/README.md:13]: Who builds these prebuilt images? Can this be added to the CI?

**Comment by ptrus** [comment]
Any plans for adding tests for these docker images? Otherwise, these will be broken every once in a while (as it was in the past).

File: docker/README.md (hunk 2 above)
Comment (matevz) [reply docker/README.md:13]: This should already be automated: https://github.com/oasisprotocol/oasis-web3-gateway/blob/main/.github/workflows/docker-emerald-dev.yml

Comment (matevz) [reply docker/README.md:13]: Oops, will add sapphire-dev too :)

**Review by kostko — APPROVED** [review]

File: docker/emerald-dev/Dockerfile [hunk 3]
```diff
@@ -41,12 +46,13 @@ RUN tar xfvz "oasis_core_${OASIS_CORE_VERSION}_linux_amd64.tar.gz" \
...
 
-# Configure Emerald ParaTime.
-RUN mkdir -p "$(dirname ${EMERALD_PARATIME})"
-ADD "https://github.com/oasisprotocol/emerald-paratime/releases/download/v${EMERALD_PARATIME_VERSION}/emerald-paratime.orc" "${EMERALD_PARATIME}.orc"
-RUN unzip "${EMERALD_PARATIME}.orc" \
-	&& mv runtime.elf "${EMERALD_PARATIME}" \
-	&& chmod a+x "${EMERALD_PARATIME}"
+# Configure the ParaTime.
+RUN mkdir -p "$(dirname ${PARATIME})"
+# TODOD
```
Comment (ptrus) [inline docker/emerald-dev/Dockerfile]: Is this still a TODO?

Comment (matevz) [reply docker/emerald-dev/Dockerfile]: Yep. Sapphire localnet was released last week while I'm still waiting for the new Emerald release.

Comment (matevz) [reply docker/emerald-dev/Dockerfile]: localnet-emerald.orc was successfully generated. I fixed the correct path now.

**Pull request merged**